app/models/spree/user_decorator.rb

```diff
@@ -76,6 +76,22 @@ def orders_by_distributor
     data_array.sort! { |a, b| b.distributed_orders.length <=> a.distributed_orders.length }
   end
 
+  def set_bill_address(address)
+    if self.bill_address
+      self.bill_address.update_attributes(address.clone.attributes)
+    else
+      self.update_attribute(:bill_address, address.clone)
+    end
+  end
+
+  def set_ship_address(address)
+    if self.ship_address
+      self.ship_address.update_attributes(address.clone.attributes)
+    else
+      self.update_attribute(:ship_address, address.clone)
+    end
+  end
+
   private
```

Comment:
The problem is if ship_address is not nil, the update_attribute method will create a new address  record and then we need to delete the old unused address. So that's why I used update_attributes method to only update the attributes of the existing address.